src/coap_blockwise.c

```diff
@@ -90,23 +90,32 @@ static void on_block_sent(struct golioth_client *client,
     golioth_sys_sem_give(ctx->sem);
 }
 
-// Function to call the application's read block callback after a successful
-// blockwise upload
-static int call_read_block_callback(struct blockwise_transfer *ctx)
+// Function to call the application's read block callback for obtaining
+// blockwise upload data
+static enum golioth_status call_read_block_callback(struct blockwise_transfer *ctx,
+                                                    size_t *block_buffer_len)
 {
     /* Do not allow user callback to directly change ctx->block_size value */
     size_t block_size = ctx->block_size;
 
-    int err = ctx->callback.read_cb(ctx->block_idx,
-                                    ctx->block_buffer,
-                                    &block_size,
-                                    &ctx->is_last,
-                                    ctx->callback_arg);
-    if (err != GOLIOTH_OK)
+    enum golioth_status status = ctx->callback.read_cb(ctx->block_idx,
+                                                       ctx->block_buffer,
+                                                       &block_size,
+                                                       &ctx->is_last,
+                                                       ctx->callback_arg);
+    if (status != GOLIOTH_OK)
     {
-        return -err;
+        return status;
+    }
+
+    if ((ctx->is_last == false && block_size != ctx->block_size) || block_size > ctx->block_size
+        || block_size == 0)
```

Comment:
Since we're breaking this line anyway, I think it's clearer to have each condition on a separate line, rather than some combined and some separate. However, clang-format may complain about my suggestion, in which case we will respect the computer overlord.
```suggestion
    if ((ctx->is_last == false && block_size != ctx->block_size)
        || block_size > ctx->block_size
        || block_size == 0)
```

Reply:
I agree with you and [I had it formatted the same way](https://github.com/golioth/golioth-firmware-sdk/compare/185aeb134836c4e464ac1b478e3c1b1ea824e785..c66399cfc006c1aa22303493d99671cbfdcb15fd) until our computer overlord showed me the true way (that is much uglier and harder to read).